ce/api/geo.py

```diff
@@ -52,88 +53,144 @@ def inner(obj, _seen_ids = None):
 log = logging.getLogger(__name__)
 cache_lock = RLock()
 
-class memoize_mask(object):
+class memoize(object):
```

Comment:
Not saying that you necessarily *should* do this (or that I'm requesting a change to this PR), but I believe that this decorator could be written more elegantly in a functional style than in an OO style. For example, I think that all of the state that you store in `self.*` in the `__init__()` functions could just as easily come from a closure context.

But this works too (in a slightly more verbose manner); just want you to be aware of the various stylistic options.